src/components/common/SafeTokenWidget/index.tsx

```diff
@@ -1,38 +1,51 @@
-import { SAFE_TOKEN_ADDRESSES } from '@/config/constants'
+import { SAFE_REACT_URL, SAFE_TOKEN_ADDRESSES } from '@/config/constants'
+import { AppRoutes } from '@/config/routes'
 import useBalances from '@/hooks/useBalances'
 import useChainId from '@/hooks/useChainId'
+import { formatAmountWithPrecision } from '@/utils/formatNumber'
 import { safeFormatUnits } from '@/utils/formatters'
 import { Box, ButtonBase, Skeleton, Typography } from '@mui/material'
+import { useRouter } from 'next/router'
 
 import SafeTokenIcon from './safe_token.svg'
 
 import css from './styles.module.css'
 
+// TODO: once listed on safe apps list, get the url from there?
+const CLAIMING_APP_URL = 'https://safe-claiming-app.pages.dev/'
+
 export const getSafeTokenAddress = (chainId: string): string => {
   return SAFE_TOKEN_ADDRESSES[chainId]
 }
 
 const SafeTokenWidget = () => {
   const balances = useBalances()
   const chainId = useChainId()
+  const router = useRouter()
 
   const tokenAddress = getSafeTokenAddress(chainId)
   if (!tokenAddress) {
     return null
   }
 
+  const shareUrl = `${SAFE_REACT_URL}/share/safe-app?appUrl=${CLAIMING_APP_URL}&chainId=${chainId}`
+  const url = router.query.safe
+    ? `${AppRoutes.safe.apps}?safe=${router.query.safe}&appUrl=${CLAIMING_APP_URL}`
+    : shareUrl
```

Comment:
I don't understand why we need to check `router.query.safe`. Can we not always link to the internal route?

Reply:
I just copy pasted it from how we render the links for our [AppCards](https://github.com/safe-global/web-core/blob/main/src/components/safe-apps/AppCard/index.tsx#L123)

Not sure why it was implemented that way.